src/posthog-surveys.ts

```diff
@@ -104,7 +121,23 @@ export class PostHogSurveys {
                     ? this.instance.featureFlags.isFeatureEnabled(survey.internal_targeting_flag_key)
                     : true
 
-                return linkedFlagCheck && targetingFlagCheck && internalTargetingFlagCheck
+                const hasEvents = survey.events && survey.events.length > 0
+                const eventBasedTargetingFlagCheck = hasEvents ? activatedSurveys.includes(survey.id) : true
+                if (eventBasedTargetingFlagCheck) {
+                    // TODO: think more of surveys that can be shown repeatedly on the same event.
+                    // an example is someone clicking a button and we want to show a survey every time
+                    // not just once.
+                    // unRegister will remove the survey from being watched.
+                    // The following mechanisms prevent us from showing the survey twice to the same user.
+                    // 1.`seenSurvey_{id}` check in local storage in the surveys.tsx file.
+                    // 2. internalTargetingFlagCheck will be false if a user has already seen a survey
+                    // need to override it here if a user can see the same survey multiple times.
+                    eventsReceiver.deRegister(survey)
```

Comment:
unclear why we're deregistering here - like you say there's other mechanisms to prevent double viewing - but assuming this is just a placeholder for the TODO.